packages/orchestration/src/chain-info.js

```diff
@@ -64,6 +65,44 @@ const knownChains = /** @satisfies {Record<string, ChainInfo>} */ (
 
 /** @typedef {typeof knownChains} KnownChains */
 
+/**
+ * @param {ERef<import('@agoric/vats').NameHubKit['nameAdmin']>} agoricNamesAdmin
+ * @param {string} name
+ * @param {CosmosChainInfo} chainInfo
+ * @param {(...messages: string[]) => void} log
+ */
+export const registerChain = async (
+  agoricNamesAdmin,
+  name,
+  chainInfo,
+  log = () => {},
+) => {
+  const { nameAdmin } = await E(agoricNamesAdmin).provideChild('chain');
+  const { nameAdmin: connAdmin } =
+    await E(agoricNamesAdmin).provideChild('chainConnection');
+
+  mustMatch(chainInfo, CosmosChainInfoShape);
+  const { connections = {}, ...vertex } = chainInfo;
+
+  const promises = [
+    E(nameAdmin)
+      .update(name, vertex)
+      .then(() => log(`registered agoricNames chain.${name}`)),
+  ];
+
+  // FIXME updates redundantly, twice per edge
+  for await (const [counterChainId, connInfo] of Object.entries(connections)) {
```

Comment:
```suggestion
  for (const [counterChainId, connInfo] of Object.entries(connections)) {
```
Is `connections` an async iterable? Seems we already have it synchronously?

Reply:
I often write `for await ...` if there are any `await`s in the loop body. Is that not necessary?

Reply:
Not here, it was a vestige. Removing.

Reply:
> I often write for await ... if there are any awaits in the loop body. Is that not necessary?

My understanding is that is has no effect. It's only necessary when iterating over an async iterable.